src/theme/NotFound.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import Translate, { translate } from '@docusaurus/Translate';
+import { PageMetadata } from '@docusaurus/theme-common';
+import Layout from '@theme/Layout';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import { useLocation } from '@docusaurus/router';
+
+interface DocEntry {
+  id: string;
+  path: string;
+  sidebar: string;
+}
+
+interface DocSection {
+  versions: Array<{
+    docs: Array<DocEntry>;
+  }>;
+}
+
+function useFindMatchingEntries(searchingLocation: string): DocEntry[] {
+  const locationPaths = searchingLocation.split('/').filter(Boolean);
+  const context = useDocusaurusContext();
+
+  const sections = context.globalData[
+    'docusaurus-plugin-content-docs'
+  ] as Record<string, DocSection>;
+
+  const entries = Object.values(sections)
+    .map((section) => {
+      const entries = section.versions[0].docs;
+      return entries;
+    })
+    .flat();
+
+  return entries.filter((entry) =>
+    locationPaths.some((part) => entry.id.includes(part)),
+  );
+}
+
+export default function NotFound() {
+  const { pathname } = useLocation();
+  const possibleMatchingEntries = useFindMatchingEntries(pathname);
+
+  return (
+    <>
+      <PageMetadata
+        title={translate({
+          id: 'theme.NotFound.title',
+          message: 'Page Not Found',
+        })}
+      />
+      <Layout>
+        <main className='container margin-vert--xl'>
+          <div className='row'>
+            <div className='col col--6 col--offset-3'>
+              <h1 className='hero__title'>
+                <Translate
+                  id='theme.NotFound.title'
+                  description='The title of the 404 page'
+                >
+                  Page Not Found
+                </Translate>
+              </h1>
+              {possibleMatchingEntries.length > 0 ? (
+                <>
+                  <p>Maybe you were looking for these:</p>
```

Comment:
I would improve this message, thoughts @Dr-Electron ?

Reply:
Maybe more like:

### This article does not exist in Shimmer

Articles related to the topic you might be interested in:

- ...
- ...

@lucas-tortora @aleksei-korolev Maybe you have thoughts about this?

Reply:
@jlvandenhout I think we could try something a bit more informal, for example:

Reply:
I like the first version of @lucas-tortora. Do we want to try to go with that? 🤔 

```suggestion
                  <p>It seems we could not find what you're looking for, but you may be interested in:</p>
```